specification/dartLangSpec.tex

```diff
@@ -14081,6 +14084,23 @@ \subsection{Property Extraction}
 (\ref{superGetterAccessAndMethodClosurization}).
 \EndCase
 
+\LMHash{}%
+\Case{Implicit}
+Let $e$ be an expression whose static type is
+an interface type that has a method named \CALL.
+In the case where the context type for $e$ is a function type,
+$e$ is treated as \code{$e$.\CALL}.
+
+\commentary{%
+This means that a ``callable object'' may be treated as a function
+that supports a mechanism similar to function closurization
```

Comment:
What is a "function closurization" here? (Words, they mean things! But what?!?)

Callable objects are usually treated like function objects, which are inherently closed, so closurization isn't necessary. So to me "function" means "already closed", but I guess it can also be used about static functions.

Reply:
Check the index: Function closurization is a concept and a section.

It's the process that takes something which is not specified to have any particular properties (a 'function') except for having a declaration, and yields a function object, whose behavior is specified to be the same as running the code in the body of said declaration with suitable bindings of formal parameters and type parameters the function declaration.

The reason why one could say that this is a mechanism which is similar to a function closurization is that we have some expression (in this case we do start with an expression whose value is actual object), and then we take some steps in order to obtain a function object that "corresponds" to the given expression.

However, it's commentary so it's always a discussion whether it's too imprecise to be helpful.

Technically, after desugaring, it is simply an instance method closurization (somehow named 'ordinary method closurization' in the spec, maybe I should fix that immediately...).

Reply:
Fixed that, too.